src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskReconfigurationService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.MarkTaskToReconfigureTaskFilter;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskFilter;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class TaskReconfigurationService {
+
+    private final CFTTaskDatabaseService cftTaskDatabaseService;
+
+    public TaskReconfigurationService(CFTTaskDatabaseService cftTaskDatabaseService) {
+        this.cftTaskDatabaseService = cftTaskDatabaseService;
+    }
+
+    @Transactional
+    public List<TaskResource> markTasksToReconfigure(List<TaskFilter<?>> taskFilters) {
+        List<String> caseIds = taskFilters.stream()
+            .filter(filter -> filter.getKey().equalsIgnoreCase("case_id"))
+            .flatMap(filter -> ((MarkTaskToReconfigureTaskFilter)filter).getValues().stream())
+            .map(Object::toString)
+            .collect(Collectors.toList());
+
+        List<TaskResource> taskResources = cftTaskDatabaseService
+            .getActiveTasksByCaseIdsAndReconfigureRequestTimeIsNull(
+                caseIds, List.of(CFTTaskState.ASSIGNED, CFTTaskState.UNASSIGNED));
+
+        taskResources.stream()
+            .map(task -> cftTaskDatabaseService.findByIdAndObtainPessimisticWriteLock(task.getTaskId())
+                .orElseGet(() -> null))
+            .filter(Objects::nonNull)
+            .forEach(taskResource -> {
+                taskResource.setReconfigureRequestTime(OffsetDateTime.now());
+                cftTaskDatabaseService.saveTask(taskResource);
+            });
```

Comment:
As discussed we should handle errors a little better.  Please add that as a requirement to the errors story so we don't miss it.

Reply:
added as requirement in RWA-1196 error scenarios story